pkgs/fc/agent/fc/manage/collect_garbage.py

```diff
@@ -28,100 +30,194 @@
+        ]
+        log.info(
+            "userscan-start",
+            _replace_msg="Running fc-userscan for {user_count} users",
+            user_count=len(users_to_scan),
+        )
 
     for user in users_to_scan:
-        log.debug(
+        home_dir = Path(user.pw_dir)
+        cache_file = home_dir / ".cache/fc-userscan.cache"
+        user_specific_ignore_file = home_dir / ".userscan-ignore"
+
+        user_log = log.bind(name=user.pw_name)
+
+        user_log.info(
             "userscan-user",
-            _replace_msg="Scanning {homedir} as {name}",
-            homedir=user.pw_dir,
-            name=user.pw_name,
+            _replace_msg="Scanning {homedir}",
+            homedir=home_dir,
         )
 
-        p = subprocess.Popen(
-            [
-                "fc-userscan",
-                "--register",
-                "--cache",
-                user.pw_dir + "/.cache/fc-userscan.cache",
-                "--cache-limit",
-                "10000000",
-                "--unzip=*.egg",
-                "--excludefrom",
-                exclude_file,
-                user.pw_dir,
-            ],
+        if cache_file.exists():
+            user_log.debug(
+                "userscan-cache-found",
+                path=cache_file,
+                size=cache_file.stat().st_size,
+                owner=cache_file.owner(),
+            )
+        else:
+            user_log.debug(
+                "userscan-cache-not-found",
+                path=cache_file,
+            )
+
+        userscan_cmd = [
+            "fc-userscan",
+            "--list" if dry_run_for_user else "--register",
+            "--cache",
+            cache_file,
+            "--cache-limit",
+            "10000000",
+            "--unzip=*.egg",
+            "--excludefrom",
+            exclude_file,
+            home_dir,
+        ]
+
+        if user_specific_ignore_file.exists():
+            user_log.debug(
+                "userscan-ignore-found",
+                path=user_specific_ignore_file,
+                size=user_specific_ignore_file.stat().st_size,
+                owner=user_specific_ignore_file.owner(),
+            )
+
+            userscan_cmd.extend(["--excludefrom", user_specific_ignore_file])
+        else:
+            user_log.debug(
+                "userscan-ignore-not-found",
+                path=user_specific_ignore_file,
+            )
+
+        if verbose:
+            userscan_cmd.append("--verbose")
+
+        user_log.debug("userscan-cmd", cmd=userscan_cmd)
+
+        proc = subprocess.Popen(
+            userscan_cmd,
             stdin=subprocess.DEVNULL,
-            preexec_fn=lambda: os.setresuid(user.pw_uid, 0, 0),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
         )
-        rc = p.wait()
-        log.debug("userscan-result", rc=rc)
-        return_codes.append(rc)
 
+        # FIXME: unused variable, will this be used in the future?
```

Comment:
@dpausp ruff complained when rebasing, please review whether this is still needed.